src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -203,6 +203,38 @@ public function findByIds(array $ids): array
             ->getQuery()->getResult(Query::HYDRATE_ARRAY);
     }
 
+    /**
+     * @param int[]    $ancestryDbIds
+     * @param string   $entityType
+     * @param int|null $limit
+     * @param int      $offset
+     * @return array
+     */
+    public function findProgrammesByAncestryAndType(array $ancestryDbIds, string $entityType, ?int $limit, int $offset) : array
```

Comment:
FindDescendantsByType might be a better name

I'd be tempted to open this up to all types now, to save you having to edit this when you add in group support later, if you do that then rename 'programme' to be 'entity' because it could be a programme or group .